games-action/another_world/additional-files/another_world.sh

```diff
@@ -22,9 +27,10 @@ if [ ! -f $DATA_PATH/BANK01 ] \
 || [ ! -f $DATA_PATH/BANK0D ] \
 || [ ! -f $DATA_PATH/MEMLIST.BIN ];
 then
-    if [ "$(alert --warning "Please copy original Another World files (DOS version) to $DATA_PATH (BANK* and MEMLIST.BIN)." "Open $DATA_PATH" "Cancel")" == "Open $DATA_PATH" ]; then
+    if [ "$(alert --warning "Please copy original 'Another World' files (DOS version) to $DATA_PATH (BANK* and MEMLIST.BIN)." "Open data folder" "Cancel")" == "Open data folder" ];
```

Comment:
Please use a for loop:

```
for file in BANK01 BANK02; do
   if [ ! -f $DATA_PATH/$file ]; then
      ...
      exit 1
   fi
done

neo-raw ...
```

Also calling a script in a Terminal shouldn't popup a graphical alert...

Reply:
hi @korli 
I agree the using a for loop is much cleaner and I'll fix this later today.

> Also calling a script in a Terminal shouldn't popup a graphical alert...

While I can certainly understand what you mean, I've used that popup in a script for the following reasons:
1) It was suggested in the forum thread [https://discuss.haiku-os.org/t/another-world-raw-ported-to-haiku/13962/9](url).
2) It was already used in (among others) JA2 recipe [https://github.com/haikuports/haikuports/blob/master/games-strategy/ja2_stracciatella/additional-files/ja2_stracciatella.sh](url) Indeed my startup script is greatly inspired from this one.
3) It is surely a great QoL for newbie (and/or lazy) users.
4) There is no other way to notify the user about the missing game files.

That said, please let me know if this is something unacceptable and I'll remove it.

Reply:
It would be possible to test for the TERM variable to be set: if it is set, you're running from Terminal, you can use echo. If it is not set, you're running from Tracker, use alerts.

But maybe we should keep this simple and stop over-engineering it?

Reply:
I thought there was something else than alert, which correctly uses a popup or the Terminal input. Maybe not.

Reply:
I like what suggested by @pulkomandy. I'll use the TERM var to choose between an echo or an alert.

Tonight I'll implement these two things. Thank you.

PS: is even possible to boot haiku in text-only mode? I wasn't aware of that.

Reply:
Yes, I never remember how it is between Haiku and BeOS (one sets it to "dumb", the other does not set it at all if I remember correctly)